### PR TITLE
Add intent-based flow to enable Bluetooth for T and above.

### PR DESCRIPTION
**Pull request by xianyuanjia**
This change is [Reviewable](https://reviewable.io/reviews/google/mobly-bundled-snippets/151)

### CODE REVIEW

File: src/main/java/com/google/android/mobly/snippet/bundled/bluetooth/BluetoothAdapterSnippet.java [hunk 1]
```diff
@@ -119,7 +119,19 @@ public void btEnable() throws BluetoothAdapterSnippetException, InterruptedExcep
         }
         waitForStableBtState();
 
-        if (!mBluetoothAdapter.enable()) {
+        if (Build.VERSION.SDK_INT > 32) {
+            // BluetoothAdapter#enable is removed from public SDK for T and above, so uses an intent
```
Comment (xpconanfan) [inline src/main/java/com/google/android/mobly/snippet/bundled/bluetooth/BluetoothAdapterSnippet.java:123]: this doesn't explain why there is a separate block for `> 30`

File: src/main/java/com/google/android/mobly/snippet/bundled/bluetooth/BluetoothAdapterSnippet.java [hunk 2]
```diff
@@ -119,7 +119,19 @@ public void btEnable() throws BluetoothAdapterSnippetException, InterruptedExcep
...
+            UiDevice uiDevice = getUiDevice();
+            Intent enableIntent = new Intent(BluetoothAdapter.ACTION_REQUEST_ENABLE);
+            enableIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+            // Triggers the system UI popup to ask for explicit permission.
+            mContext.startActivity(enableIntent);
+            // Clicks the "ALLOW" button.
+            BySelector allowButtonSelector = By.text(TEXT_PATTERN_ALLOW).clickable(true);
+            uiDevice.wait(Until.findObject(allowButtonSelector), 10);
+            uiDevice.findObject(allowButtonSelector).click();
+        } else if (!mBluetoothAdapter.enable()) {
```
Comment (xpconanfan) [inline src/main/java/com/google/android/mobly/snippet/bundled/bluetooth/BluetoothAdapterSnippet.java:134]: is this `else` correct?

**Review by xpconanfan — CHANGES_REQUESTED** [review]

**Review by xianyuanjia — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/mobly-bundled-snippets/151)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @xpconanfan)

___
*[`src/main/java/com/google/android/mobly/snippet/bundled/bluetooth/BluetoothAdapterSnippet.java` line 123 at r1](https://reviewable.io/reviews/google/mobly-bundled-snippets/151#-N2JmTNpWXKKdM7RXz_H-r1-123:-N2NCKyc378PctePxKDF:bf7mbi9) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/069fc58e424be4fa6a668940491f0c68a6fa869c/src/main/java/com/google/android/mobly/snippet/bundled/bluetooth/BluetoothAdapterSnippet.java#L123)):*
<details><summary><i>Previously, xpconanfan (Ang Li) wrote…</i></summary><blockquote>

this doesn't explain why there is a separate block for `> 30`
</blockquote></details>

The `> 30` block is in another method, where I updated a comment. Splitting into another change.
___
*[`src/main/java/com/google/android/mobly/snippet/bundled/bluetooth/BluetoothAdapterSnippet.java` line 134 at r1](https://reviewable.io/reviews/google/mobly-bundled-snippets/151#-N2JmTNpWXKKdM7RXz_H-r1-134:-N2NCmUJ-sa1O4SPKTRy:b678bo8) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/069fc58e424be4fa6a668940491f0c68a6fa869c/src/main/java/com/google/android/mobly/snippet/bundled/bluetooth/BluetoothAdapterSnippet.java#L134)):*
<details><summary><i>Previously, xpconanfan (Ang Li) wrote…</i></summary><blockquote>

is this `else` correct?
</blockquote></details>

It calls the API directly only if the API level is not `> 32`

**Review by xpconanfan — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/google/mobly-bundled-snippets/151#-:-N2O3-qqBDmXW90q3ed9:bnfp4nl)* status: 0 of 1 files reviewed, all discussions resolved (waiting on @xpconanfan)

**Review by xpconanfan — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/mobly-bundled-snippets/151)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @xianyuanjia and @xpconanfan)

___
*[`src/main/java/com/google/android/mobly/snippet/bundled/bluetooth/BluetoothAdapterSnippet.java` line 122 at r2](https://reviewable.io/reviews/google/mobly-bundled-snippets/151#-N2O3KP-BgYJ-KQGUAq0:-N2O3KP-BgYJ-KQGUAq1:b-ippc1f) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/66da8196e9087bb24c5919e7ebcd9d4217ea8b6a/src/main/java/com/google/android/mobly/snippet/bundled/bluetooth/BluetoothAdapterSnippet.java#L122)):*
> ```Java
>         waitForStableBtState();
> 
>         if (Build.VERSION.SDK_INT > 32) {
> ```

let's use `>=33` so it's clearer
___
*[`src/main/java/com/google/android/mobly/snippet/bundled/bluetooth/BluetoothAdapterSnippet.java` line 123 at r2](https://reviewable.io/reviews/google/mobly-bundled-snippets/151#-N2O3ElR-WrD_z1JNZat:-N2O3ElS70fcD6QOAqUa:b-8jo373) ([raw file](https://github.com/google/mobly-bundled-snippets/blob/66da8196e9087bb24c5919e7ebcd9d4217ea8b6a/src/main/java/com/google/android/mobly/snippet/bundled/bluetooth/BluetoothAdapterSnippet.java#L123)):*
> ```Java
> 
>         if (Build.VERSION.SDK_INT > 32) {
>             // BluetoothAdapter#enable is removed from public SDK for T and above, so uses an intent
> ```

let's use the number system instead of letters in comments

**Review by xpconanfan — APPROVED** [review]

**Pull request merged**